src/persist-types/src/lib.rs

```diff
@@ -50,3 +50,34 @@ pub trait Codec: Sized + 'static {
     // without any copies, see if we can make the types work out for that.
     fn decode<'a>(buf: &'a [u8]) -> Result<Self, String>;
 }
+
+/// Encoding and decoding operations for a type usable as a persisted timestamp
+/// or diff.
+//
+// WIP: Figure out the requirements here. E.g. if we can require timestamps and
+// diffs be roundtrip-able as 8 bytes, persist can do some nice performance
+// things in its internal columnar storage format compared to if they were tied
```

Comment:
For my curiosity, what can we optimize if ts and diffs serialize to 64 bits? I can't remember exactly (I can find the slack thread if useful) but we had one usecase where we wanted to persist a diff that was something complicated and we ended up doing a bunch of work to make look like a collection because the diffs were hardcoded to `isize`.

Could whatever optimization we get from this being 8bytes implemented by having a `hint` method on the generic `Codec` trait?

Reply:
two things:
- it's a columnar format, so if you have fixed sized things (and the schema knows they're fixed size) then you can avoid encoding the lengths of them
- inside parquet (the concrete columnar format we use), a bunch of of the compression algorithms are only available for columns of integral types and not for columns of byte arrays

> Could whatever optimization we get from this being 8bytes implemented by having a hint method on the generic Codec trait?

I don't think but perhaps I'm missing something